sdk/highlightinc-highlight-datasource/pkg/plugin/datasource.go

```diff
@@ -196,6 +223,66 @@ func (d *Datasource) CallResource(ctx context.Context, req *backend.CallResource
 		if err != nil {
 			return err
 		}
+
+	case "traces-key-values":
+		var q struct {
+			TracesKeyValues []QueryKeyValue `graphql:"traces_key_values(project_id: $project_id, date_range: $date_range, key_name: $query)"`
```

Comment:
I think it would be nice to use the polymorphic `key_values` resolver to avoid most of the code in each of these switch cases. The above code could also be refactored to use the `keys` resolver. I think both of those polymorphic resolvers were added after this original code